src/lib/logs/log_forwarder.py

```diff
@@ -51,11 +54,18 @@ async def run_logs(logging_context: LoggingContext, instance_metadata: InstanceM
 
     # initiate http client for both GCP and DT
     async with init_gcp_client_session() as gcp_session:
-        gcp_client = GCPClient(await create_token(logging_context, gcp_session))
+        token = await create_token(logging_context, gcp_session)
+        if token is None:
+            raise Exception("Cannot start pubsub streaming pull - 'Failed to fetch token")
```

Comment:
streaming pull has specific meaning in type of pulling from pub/sub. Since we are not using streaming pull method, I would suggest to change the naming.